independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapMavenContext.java

```diff
@@ -533,7 +533,18 @@ private DefaultRepositorySystemSession newRepositorySystemSession() throws Boots
         DefaultSettingsDecryptionRequest decrypt = new DefaultSettingsDecryptionRequest();
         decrypt.setProxies(settings.getProxies());
         decrypt.setServers(settings.getServers());
+        // need to set `settings-security.xml` location extra, because it isn't discovered
+        // by BeanBag when constructing `DefaultSecDispatcher`
+        File settingsSecurityXml = null;
+        boolean setSettingsSecurity = !System.getProperties().contains(SETTINGS_SECURITY)
```

Comment:
```suggestion
        boolean setSettingsSecurity = !System.getProperties().containsKey(SETTINGS_SECURITY)
```

Otherwise it maps to the value

Reply:
Great point! I didn't even bother checking the code I was restoring. Shame on me :-)

Reply:
Fixed, thanks!